### PR TITLE
[tree.hh] Add new port. 

### PR DESCRIPTION
**Pull request by Hconk**
**Add a new port tree.hh**
An STL-like C++ header-only tree library

- #### What does your PR fix?  
   #19799

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  all, No

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes 

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/treehh/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,10 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO kpeeters/tree.hh
+    REF 8dc57bc2bb7cd2fd1d6e777c6b5b58db0c3852ef
+    SHA512 a40481952802fdfab05159822b012a576ea956135a44f88d93844cb578c49ed5fb6fb7330bce699d7fa6ba1da65d12b284b6e9e6b103eadfd76cbc96d6839db5
+    HEAD_REF master
+)
+
+file(INSTALL ${SOURCE_PATH}/src/tree.hh DESTINATION ${CURRENT_PACKAGES_DIR}/include)
+file(INSTALL ${SOURCE_PATH}/src/tree.hh DESTINATION ${CURRENT_PACKAGES_DIR}/share/treehh RENAME copyright)
```
Comment (PhoebeHui) [inline ports/treehh/portfile.cmake:10]:
```suggestion
file(INSTALL "${SOURCE_PATH}/src/tree.hh" DESTINATION "${CURRENT_PACKAGES_DIR}/include")
file(INSTALL "${SOURCE_PATH}/src/tree.hh" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)

```

Comment (PhoebeHui) [reply ports/treehh/portfile.cmake:10]: @Hconk, Could you minimize the copyright? please only include the  license information instead.

@BillyONeal, do we need install tree.hh into a subidrectory like treehh/tree.hh?

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
@Hconk, thanks for your PR!

File: ports/treehh/portfile.cmake (hunk 1 above)
Comment (Hconk) [reply ports/treehh/portfile.cmake]: Okay, move tree.hh to treehh/tree.hh , and minimize the copyright.

**Review by PhoebeHui — APPROVED** [review]
Thanks for your updates!

**Comment by BillyONeal** [comment]
Thanks for the new port!

**Pull request merged**